File: 2566. Maximum Difference by Remapping a Digit/Solution.cpp
#include <algorithm>
#include <ranges>
#include <string>
using namespace std;

class Solution {
public:
    int minMaxDifference(int num)
    {
        string s = to_string(num);
        int mx = num;
        for (char c : s) {
            if (c != '9') {
                string tmp = s;
                ranges::replace(tmp, c, '9');
                mx = stoi(tmp);
                break;
            }
        }

        char c0 = s[0];
        ranges::replace(s, c0, '0');
        int mn = stoi(s);
        return mx - mn;
    }
};

File: 2566. Maximum Difference by Remapping a Digit/Solution.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        s = str(num)
        mx = num
        for c in s:
            if c != "9":
                mx = int(s.replace(c, "9"))
                break
        mn = int(s.replace(s[0], "0"))
        return mx - mn
